golang/cosmos/app/app.go

```diff
@@ -803,14 +804,88 @@ func NewAgoricApp(
 	return app
 }
 
+type swingStoreMigrationEventHandler struct {
+	swingStore sdk.KVStore
+}
+
+func (eventHandler swingStoreMigrationEventHandler) OnExportStarted(height uint64, retrieveSwingStoreExport func() error) error {
+	return retrieveSwingStoreExport()
+}
+
+func (eventHandler swingStoreMigrationEventHandler) OnExportRetrieved(provider swingsetkeeper.SwingStoreExportProvider) (err error) {
+	defer helpers.RecoverToError(&err)
+
+	var exportDataIterator sdk.Iterator
+	exportDataIterator, err = provider.GetExportDataIterator()
```

Comment:
This use of `helpers.RecoverToError` probably warrants an explanatory comment, which would then remove the need to specifically declare `exportDataIterator`.
```suggestion
	// Work around sdk.Iterator abuse of `panic`.
	defer helpers.RecoverToError(&err)
	exportDataIterator, err := provider.GetExportDataIterator()
```

Reply:
Removed after switching to KVEntryReader